services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/service/implement/SubnetServiceImp.java

```diff
@@ -238,6 +239,16 @@ public IpAddrRequest allocateIpAddressForGatewayPort(String subnetId, String cid
         if (highIp == null || lowIp == null) {
             return null;
         }
+
+        String[] highIps = highIp.split("\\.");
```

Comment:
Should we add the logic here? 

cidrToFirstIpAndLastIp is used somewhere else that actually asks the first and last Ip, right? If so, then this new logic is going to break codes in those places.

I think we should have a new method that handles the cut explicitly.

Reply:
This method is used for creating Ip Address Range, and Ip Address Range may also need this fix, right? let us talk about this.